src/partners/thorchain.ts

```diff
@@ -52,169 +61,247 @@ const asThorchainResult = asObject({
+          )
+          return match
+        })
 
-      const depositAmount =
-        Number(txIns[0].coins[0].amount) / THORCHAIN_MULTIPLIER
+        // If there is a match between source and dest asset that means a refund was made
+        // and the transaction failed
+        if (srcDestMatch) {
+          continue
+        }
 
-      const txOut = txOuts.find(o => o.coins[0].asset !== 'THOR.RUNE')
-      if (txOut == null) {
-        continue
-      }
+        const timestampMs = div(date, '1000000', 16)
+        const { timestamp, isoDate } = smartIsoDateFromTimestamp(
+          Number(timestampMs)
+        )
+
+        const [chainAsset] = srcAsset.split('-')
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const [_chain, asset] = chainAsset.split('.')
+
+        // Find the first output that does not match the affiliate address
+        // as this is assumed to be the true destination asset/address
+        // If we can't find one, then just match the affiliate address as
+        // this means the affiliate address is the actual destination.
+        const hasAffiliateFlag = txOuts.some(o => o.affiliate === true)
+        let txOut = txOuts.find(out => {
+          if (hasAffiliateFlag) {
+            return out.affiliate !== true
+          } else {
+            return out.address !== thorchainAddress
+          }
+        })
+        if (txOut == null) {
+          if (txOuts.length !== 1) {
+            throw new Error(
+              `${pluginId}: Unexpected ${txOuts.length} txOuts. Expected 1`
+            )
+          }
+          txOut = txOuts[0]
+        }
 
-      let payoutCurrency
-      let payoutAmount = 0
-      if (txOut != null) {
         const [destChainAsset] = txOut.coins[0].asset.split('-')
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
         const [_destChain, destAsset] = destChainAsset.split('.')
-        payoutCurrency = destAsset
-        payoutAmount = Number(txOut.coins[0].amount) / THORCHAIN_MULTIPLIER
-      }
+        const payoutCurrency = destAsset
+        const payoutAmount =
+          Number(txOut.coins[0].amount) / THORCHAIN_MULTIPLIER
 
-      const ssTx: StandardTx = {
-        status,
-        orderId: txIns[0].txID,
-        depositTxid: txIns[0].txID,
-        depositAddress: undefined,
-        depositCurrency: asset.toUpperCase(),
-        depositAmount,
-        payoutTxid: txOut?.txID,
-        payoutAddress: txOut?.address,
-        payoutCurrency,
-        payoutAmount,
-        timestamp,
-        isoDate,
-        usdValue: -1,
-        rawTx
-      }
+        const ssTx: StandardTx = {
+          status,
+          orderId: txIns[0].txID,
+          depositTxid: txIns[0].txID,
+          depositAddress: undefined,
+          depositCurrency: asset.toUpperCase(),
+          depositAmount,
+          payoutTxid: txOut?.txID,
+          payoutAddress: txOut?.address,
```

Comment:
Elvis operators aren't needed anymore because the type has narrowed to exclude `undefined`.